traitlets/traitlets.py

```diff
@@ -418,6 +420,10 @@ def _dynamic_default_callable(self, obj):
         return getattr(self, 'make_dynamic_default', None)
 
     def instance_init(self, obj):
+        if not hasattr(obj, '_trait_values'):
```

Comment:
I guess redundant is fine here. I am merely removing complexity from the base class' `__new__`, and I prefer not adding more magic than the `instance_init`.

I could also simply have put it at the top of `HasTraits::__init__` but this would make things such as the following thing not work
```Python
class Foo(HasTraits):
    def __init__(self, **kwargs):
        self.observe(...)   # call some observe before super.__init__, which requires _trait_values
        super(Foo, self).__init__(**kwargs)
```

Then, there is nothing left that is trait-specific in `HasDescriptors`.